### PR TITLE
Make the UI of Register screen uniform #8 Issue Solved #25

### PR DESCRIPTION
**Pull request by raviyadav17**
1. Solved the issue #8 
2. Updated the Register.dart Page.
3. Refactor the Input Text fields.

I am a part of Codepeak. Thank You

### CODE REVIEW

File: lib/Screens/Register.dart [hunk 1]
```diff
@@ -43,14 +43,172 @@ class _RegisterState extends State<Register> {
...
+  bool hidePass1 = true;
+  bool hidePass2 = true;
   String userImageUrl = '';
   File _imageFile;
 
 
   @override
+
+
+  Widget Template1(String hint , String error , Icon icon){
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:58]: Give a meaningful name, please!

File: lib/Screens/Register.dart [hunk 2]
```diff
@@ -43,14 +43,172 @@ class _RegisterState extends State<Register> {
...
+
+          ),
+        ),
+        SizedBox(height: 20,)
+      ],
+    );
+  }
+
+
+  Widget Template2(String hint , String error ) {
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:98]: Ditto!

File: lib/Screens/Register.dart [hunk 3]
```diff
@@ -43,14 +43,172 @@ class _RegisterState extends State<Register> {
...
+              decoration: InputDecoration(
+                hintText: "$hint",
+                prefixIcon: Icon(Icons.vpn_key),
+                suffixIcon: IconButton(
+                    icon: Icon(
+                      hidePass1
+                          ? Icons.visibility
+                          : Icons.visibility_off,
+                      //color: Theme.of(context).primaryColorDark,
+                      //Icons.remove_red_eye
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:135]:
```suggestion
```

File: lib/Screens/Register.dart [hunk 4]
```diff
@@ -43,14 +43,172 @@ class _RegisterState extends State<Register> {
...
+                      });
+                    }),
+              ),
+              validator: (val) =>
+              val.length < 6
+                  ? '$error'
+                  : null,
+              textAlignVertical: TextAlignVertical.bottom,
+              obscureText: hidePass1,
+              // net ninja
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:149]:
```suggestion
```

File: lib/Screens/Register.dart [hunk 5]
```diff
@@ -43,14 +43,172 @@ class _RegisterState extends State<Register> {
...
+                decoration: InputDecoration(
+                  hintText: "$hint",
+                  prefixIcon: Icon(Icons.vpn_key),
+                  suffixIcon: IconButton(
+                      icon: Icon(
+                        hidePass2
+                            ? Icons.visibility
+                            : Icons.visibility_off,
+                        //color: Theme.of(context).primaryColorDark,
+                        //Icons.remove_red_eye
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:188]:
```suggestion
```

File: lib/Screens/Register.dart [hunk 6]
```diff
@@ -144,85 +304,147 @@ class _RegisterState extends State<Register> {
...
-                                //onChanged: (val) {
-                                //setState(() {
-                                //password = val;
-                                //});
-                                //},
+                              ),*/
+
+                        Template2("Enter Password", 'Enter a password 6+ chars long'),
+
+                        /* Container(
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:358]: I think we can get rid of this now?

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @raviyadav17! Looks good! Here is a high-level review, I'll look into this more soon!

File: lib/Screens/Register.dart (hunk 1 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Done

File: lib/Screens/Register.dart (hunk 2 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Done

File: lib/Screens/Register.dart (hunk 3 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Removed This

File: lib/Screens/Register.dart (hunk 5 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Removed This

File: lib/Screens/Register.dart (hunk 6 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Removed This

File: lib/Screens/Register.dart (hunk 4 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart]: Removed It.

**Comment by Saransh-cpp** [comment]
Thanks, @raviyadav17! This looks great! Can you merge in master before the final review? Your branch is 88 commits behind my repository's master branch. Let me know if you need any help!

**Comment by raviyadav17** [comment]
Successfully Merged Update Register.dart branch to master branch.

File: lib/Screens/Register.dart [hunk 7]
```diff
@@ -43,14 +43,167 @@ class _RegisterState extends State<Register> {
...
+
+          ),
+        ),
+        SizedBox(height: 20,)
+      ],
+    );
+  }
+
+
+  Widget InputFieldType2(String hint , String error ) {
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:98]:
```suggestion
  Widget PasswordField(String hint, String error ) {
```

File: lib/Screens/Register.dart [hunk 8]
```diff
@@ -67,162 +220,77 @@ class _RegisterState extends State<Register> {
...
+                        //       ? Icon(Icons.add_photo_alternate)
+                        //       : NetworkImage(user.userModel.userImage),
+                        // ),
+                        // SizedBox(
+                        //   height: 20,
+                        // ),
+                        InputFieldType1("Enter Name" , 'Enter a name' , Icon(Icons.person)),
+                        InputFieldType1("Enter Email" , 'Enter an email' , Icon(Icons.email)),
+
+                        InputFieldType2("Enter Password", 'Enter a password 6+ chars long'),
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:255]:
```suggestion
                        PasswordField("Enter Password", 'Enter a password 6+ chars long'),
```

File: lib/Screens/Register.dart [hunk 9]
```diff
@@ -67,162 +220,77 @@ class _RegisterState extends State<Register> {
...
+                        InputFieldType2("Confirm Password", 'Enter a password 6+ chars long'),
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:256]:
```suggestion
                        PasswordField("Confirm Password", 'Enter a password 6+ chars long'),
```

File: lib/Screens/Register.dart [hunk 10]
```diff
@@ -43,14 +43,167 @@ class _RegisterState extends State<Register> {
...
+  bool hidePass1 = true;
+  bool hidePass2 = true;
   String userImageUrl = '';
   File _imageFile;
 
 
   @override
+
+
+  Widget InputFieldType1(String hint , String error , Icon icon){
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:58]:
```suggestion
  Widget EmailAndNameField(String hint, String error, Icon icon){
```

File: lib/Screens/Register.dart [hunk 11]
```diff
@@ -67,162 +220,77 @@ class _RegisterState extends State<Register> {
...
+                        // CircleAvatar(
+                        //   radius: 70,
+                        //   backgroundImage: user.userModel.userImage == ''
+                        //       ? Icon(Icons.add_photo_alternate)
+                        //       : NetworkImage(user.userModel.userImage),
+                        // ),
+                        // SizedBox(
+                        //   height: 20,
+                        // ),
+                        InputFieldType1("Enter Name" , 'Enter a name' , Icon(Icons.person)),
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:252]:
```suggestion
                        EmailAndNameField("Enter Name", 'Enter a name', Icon(Icons.person)),
```

File: lib/Screens/Register.dart [hunk 12]
```diff
@@ -67,162 +220,77 @@ class _RegisterState extends State<Register> {
...
+                        InputFieldType1("Enter Email" , 'Enter an email' , Icon(Icons.email)),
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:253]:
```suggestion
                        EmailAndNameField("Enter Email", 'Enter an email', Icon(Icons.email)),
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Awesome work! This is the final review, some minor naming issues, you can commit them directly from GitHub -

**Comment by Saransh-cpp** [comment]
@all-contributors please add @raviyadav17 for code, design.

File: lib/Screens/Register.dart (hunk 7 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:98]: done

File: lib/Screens/Register.dart (hunk 9 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:256]: Done

File: lib/Screens/Register.dart (hunk 8 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:255]: Done

File: lib/Screens/Register.dart (hunk 10 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:58]: Done

File: lib/Screens/Register.dart (hunk 11 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:252]: Done

File: lib/Screens/Register.dart (hunk 12 above)
Comment (raviyadav17) [reply lib/Screens/Register.dart:253]: Done

**Comment by raviyadav17** [comment]
Successfully updated the minor issues

**Comment by Saransh-cpp** [comment]
@raviyadav17 I don't see any new commits? Did you forget to push?

**Comment by raviyadav17** [comment]
> @raviyadav17 I don't see any new commits? Did you forget to push?

i updated in the master branch of my forked repo wait let me update it on other branch also

**Review by Saransh-cpp — APPROVED** [review]

**Pull request merged**